src/Components/Server/src/Circuits/CircuitRootComponentOperationBatch.cs

```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Server;
+
+internal sealed class CircuitRootComponentOperationBatch(long batchId, CircuitRootComponentOperation[] operations)
+{
+    public long BatchId => batchId;
+
+    public CircuitRootComponentOperation[] Operations => operations;
```

Comment:
I don't 100% understand why we have three independent representations for similar information:

 * `CircuitRootComponentOperationBatch`
 * `RootComponentOperationBatch` (shared source)
 * `OperationDescriptorBatch` (in WebAssembly)

If these were able to be consolidated into a single representation, or some kind of type hierarchy, it may be easier to follow. But please don't feel compelled to do a wide refactoring for this. If having independent types is a lot simpler than inheritance then it's fine.

Reply:
Agree! I just unified each of these representations by removing the Server/WebAssembly-specific types.